conditions.go

```diff
@@ -88,14 +89,46 @@ func (a Address) Equals(b Address) bool {
 // MarshalJSON provides a hex representation for JSON,
 // to override the standard base64 []byte encoding
 func (a Address) MarshalJSON() ([]byte, error) {
-	return marshalHex(a)
+	s := strings.ToUpper(hex.EncodeToString(a))
+	return json.Marshal(s)
 }
 
-// UnmarshalJSON parses JSON in hex representation,
-// to override the standard base64 []byte encoding
-func (a *Address) UnmarshalJSON(src []byte) error {
-	dst := (*[]byte)(a)
-	return unmarshalHex(src, dst)
+func (a *Address) UnmarshalJSON(raw []byte) error {
+	var enc string
+	if err := json.Unmarshal(raw, &enc); err != nil {
+		return errors.Wrap(err, "cannot decode json")
+	}
+
+	// If the encoded string starts with a prefix, cut it off and use
+	// specified decoding method instead of default one.
+	chunks := strings.SplitN(enc, ":", 2)
+	format := chunks[0]
+	if len(chunks) == 1 {
+		format = "hex"
+	} else {
+		enc = chunks[1]
+	}
+
+	// No value zero the address.
+	if len(enc) == 0 {
+		*a = nil
+		return nil
+	}
+
+	switch format {
+	case "hex":
+		val, err := hex.DecodeString(enc)
+		if err != nil {
+			return errors.Wrap(err, "cannot decode hex")
+		}
+		*a = val
+		return nil
+	case "cond":
+		*a = Condition(enc).Address()
```

Comment:
I think you need to do more parsing here

Reply:
You mean to ensure that the format is correct?
I will add a regular expression to enforce the format :muscle:

Reply:
Adressed in https://github.com/iov-one/weave/pull/365/commits/0275b0c008e2b7403a955b9103d445774cdd5850